protostar/starknet/cheaters/__init__.py

```diff
@@ -0,0 +1,16 @@
+from dataclasses import dataclass
+
+from typing_extensions import Self
+
+from .block_info import BlockInfoCheater
+
+
+@dataclass(frozen=True)
+class Cheaters:
+    block_info: BlockInfoCheater
+
+    def copy(self) -> Self:
+        return Cheaters(block_info=self.block_info.copy())
```

Comment:
We might want to make the generalization easier here, so we don't have to add new bits here every time we add a cheater

Reply:
I had a working code for `copy` and `apply` which used a combination of `dir()`, `getattr` and `setattr` but reverted it back because I sounded to me overly complicated at this step.

I have also thought about avoiding having a class that knows about all cheaters, and having a system which did `give me a cheater of this type`; but dynamic auto initialization of cheaters is not straightforward in this case because they take arbitrary dependencies from Starknet internals, like the `BlockInfo` thingy done here.